### ISSUE TITLE
No value when observing a signal Swfit 2.0

### ISSUE THREAD

**Issue by foffer** [ReactiveCocoa/ReactiveCocoa#2097]
With the following signal:
`let (daySignal, daySink) = Signal<String, NoError>.pipe()`
and with the following 'observation' of the signal:
```
dayDidChangeSignal!.observe {
    day in // This return an empty function, see attached screenshot
    self.setNewDay(day)
}
```
![screen shot 2015-06-11 at 08 42 33](https://cloud.githubusercontent.com/assets/1674650/8102170/dabaa9dc-1015-11e5-8274-ba1c1a042ca9.png)

It might be that I am doing something wrong, and that the new Swift 2.0 syntax has changed, but it doesn't seem right to me?

**Comment by foffer** [comment]
inserting `next` did the trick:

dayDidChangeSignal!.observe(next: {
    day in // This return an empty function, see attached screenshot
    self.setNewDay(day)
})

**Comment by jspahrsummers** [comment]
Looks like `next` got reordered in #2088, meaning that you're now providing the `interrupted` closure instead.

Does that sound right @neilpa?

**Comment by neilpa** [comment]
Yea. I had issues in a couple places when upgrading where the compiler got confused between the two `observe` overloads. I tweaked the params to make the issues easier to track down but forgot to switch it back. Thanks for the report.

**Comment by neilpa** [comment]
Although, I will add that I don't recommend using the trailing closure syntax for methods taking more than one closure param. Especially those like this where they all have default params.

**Comment by neilpa** [comment]
This is fixed in 64e3c2a which I've pushed to #2088
